microsoft/testsuites/network/networksettings.py

```diff
@@ -560,78 +532,144 @@ def validate_device_msg_level_change(self, node: Node, log: Logger) -> None:
         ),
     )
     def validate_device_statistics(self, environment: Environment, log: Logger) -> None:
+        # regex for filtering per vmbus channel stats from the full device statistics.
+        # [{'name': 'tx_scattered', 'value': '0'},
+        #  {'name': 'tx_no_memory', 'value': '0'},
+        #  {'name': 'tx_no_space', 'value': '0'},
+        #  {'name': 'tx_too_big', 'value': '0'},
+        #  {'name': 'vf_rx_packets', 'value': '228'},
+        #  {'name': 'vf_rx_bytes', 'value': '1158450'},
+        #  {'name': 'vf_tx_packets', 'value': '966'},
+        #  {'name': 'tx_queue_0_packets', 'value': '0'},
+        #  {'name': 'tx_queue_0_bytes', 'value': '0'},
+        #  {'name': 'rx_queue_0_packets', 'value': '91'},
+        #  {'name': 'rx_queue_0_bytes', 'value': '28378'},
+        #  {'name': 'rx_queue_0_xdp_drop', 'value': '0'},
+        #  {'name': 'tx_queue_1_packets', 'value': '0'},
+        #  {'name': 'tx_queue_1_bytes', 'value': '0'},
+        #  {'name': 'rx_queue_1_packets', 'value': '1108'},
+        #  {'name': 'rx_queue_1_bytes', 'value': '1415269'},
+        #  {'name': 'rx_queue_1_xdp_drop', 'value': '0'},
+        #  {'name': 'tx_queue_2_packets', 'value': '0'},]
+        #  {'name': 'cpu0_vf_rx_packets', 'value': '0'},]
+        #  {'name': 'cpu0_vf_rx_bytes', 'value': '0'},]
+        #  {'name': 'cpu0_vf_tx_packets', 'value': '0'},]
+        #  {'name': 'cpu0_vf_tx_bytes', 'value': '0'},]
+        tx_queue_stats_regex = re.compile(r"tx_queue_(?P<name>[\d]+)_packets")
```

Comment:
Why it's moved into the method? It means the reg expressions will be compiled every time.

Reply:
So earlier it was in the Testsuite class but then you mentioned that we should not be using self to access it. Hmm not sure then how would I use it in the methods

Reply:
Sorry, my bad. I haven't realized they are regex. I thought they are some dicts to hold intermedia data.

Reply:
The regex should be outside of a method.